ncm-metaconfig/src/main/metaconfig/kafka/pan/schema.pan

```diff
@@ -2,26 +2,198 @@ declaration template metaconfig/kafka/schema;
+    "socket.request.max.bytes" ? long(1..)
+    "socket.send.buffer.bytes" ? long(0..)
+    "transaction.max.timeout.ms" ? long(1..)
+    "transaction.state.log.load.buffer.size" ? long(1..)
+    "transaction.state.log.min.isr" ? long(1..)
+    "transaction.state.log.num.partitions" ? long(1..)
+    "transaction.state.log.replication.factor" ? long(1..)
+    "transaction.state.log.segment.bytes" ? long(1..)
+    "transactional.id.expiration.ms" ? long(1..)
+    "unclean.leader.election.enable" ? choice("true", "false")
     "zookeeper.connection.timeout.ms" ? long(100..)
-};
\ No newline at end of file
+    "zookeeper.max.in.flight.requests" ? long(1..)
+    "zookeeper.session.timeout.ms" ? long(0..)
+    "zookeeper.set.acl" ? boolean
+    "broker.id.generation.enable" ? choice("true", "false")
+    "broker.rack" ? string
+    "connections.max.idle.ms" ? long(0..)
+    "connections.max.reauth.ms" ? long(0..)
+    "controlled.shutdown.enable" ? choice("true", "false")
+    "controlled.shutdown.max.retries" ? long(0..)
+    "controlled.shutdown.retry.backoff.ms" ? long(0..)
+    "controller.socket.timeout.ms" ? long(0..)
+    "default.replication.factor" ? long(0..)
+    "delegation.token.expiry.time.ms" ? long(1..)
+    "delegation.token.master.key" ? string  # password
+    "delegation.token.max.lifetime.ms" ? long(1..)
+    "delete.records.purgatory.purge.interval.requests" ? long(0..)
+    "fetch.purgatory.purge.interval.requests" ? long(0..)
+    "group.initial.rebalance.delay.ms" ? long(0..)
+    "group.max.session.timeout.ms" ? long(0..)
+    "group.max.size" ? long(1..)
+    "group.min.session.timeout.ms" ? long(0..)
+    "inter.broker.listener.name" ? string
+    "inter.broker.protocol.version" ? string
+    "log.cleaner.backoff.ms" ? long(0..)
+    "log.cleaner.dedupe.buffer.size" ? long(0..)
+    "log.cleaner.delete.retention.ms" ? long(0..)
+    "log.cleaner.enable" ? choice("true", "false")
+    "log.cleaner.io.buffer.load.factor" ? double
+    "log.cleaner.io.buffer.size" ? long(0..)
+    "log.cleaner.io.max.bytes.per.second" ? double
+    "log.cleaner.max.compaction.lag.ms" ? long(0..)
+    "log.cleaner.min.cleanable.ratio" ? double
+    "log.cleaner.min.compaction.lag.ms" ? long(0..)
+    "log.cleaner.threads" ? long(0..)
+    "log.cleanup.policy" ? choice("compact", "delete")
+    "log.index.interval.bytes" ? long(0..)
+    "log.index.size.max.bytes" ? long(0..)
+    "log.message.format.version" ? string
+    "log.message.timestamp.difference.max.ms" ? long(0..)
+    "log.message.timestamp.type" ? string
+    "log.preallocate" ? boolean
+    "log.retention.check.interval.ms" ? long(1..)
+    "max.connections" ? long(0..)
+    "max.connections.per.ip" ? long(0..)
+    "max.connections.per.ip.overrides" ? string
+    "max.incremental.fetch.session.cache.slots" ? long(0..)
+    "num.partitions" ? long(1..)
+    "password.encoder.old.secret" ? string  # password
+    "password.encoder.secret" ? string  # password
```

Comment:
At our site profiles are public, although I know that might not be the case elsewhere. Is it safe to allow people to put passwords in profiles?

Reply:
@ned21 you mean the templates are public, not the actual profiles? there is lots of secret stuff in our profiles, but nothing in our templates.
profiles on the hosts can be restricted so not everyone can query them. we do that, only wheel group has access to them (you can set any group).